lib/tapioca/runtime/loader.rb

```diff
@@ -53,9 +53,13 @@ def rails_engines
         return [] unless Object.const_defined?("Rails::Engine")
 
         safe_require("active_support/core_ext/class/subclasses")
+        root = File.expand_path(".")
 
         # We can use `Class#descendants` here, since we know Rails is loaded
-        Object.const_get("Rails::Engine").descendants.reject(&:abstract_railtie?)
+        Object.const_get("Rails::Engine")
+          .descendants
+          .reject(&:abstract_railtie?)
+          .reject { |engine| File.fnmatch?("#{root}/**", engine.config.root) }
```

Comment:
I am afraid this is not enough to check if a gem lives under the application or not. The complications are for gems that are installed under the `vendor` folder inside the application folder, for example.

`Tapioca::Gemfile::GemSpec` class has the method `gem_in_app_dir?`, for example, that checks if any gem files are inside the directory that contains the `Gemfile`, which would also be appropriate here. That method explicitly excludes cases where a given path is under the `Bundler.bundle_path` or `Bundler.app_cache` folders.